### PR TITLE
Feat(kclvm): add linking args for cl.exe on windows and add build.bat   for build kclvm on windows.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

This PR will build the windows version of kclvm.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: samples/hello_invalid.k [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name = "kcl"
```
Comment (Peefy) [inline samples/hello_invalid.k:1]: Do not add illegal examples to examples.

File: kclvm/runner/src/linker.rs [hunk 2]
```diff
@@ -17,43 +17,3 @@ impl KclvmLinker {
         cmd.link_libs_with_cc(&lib_paths, &lib_path)
     }
 }
-
-#[allow(dead_code)]
```
Comment (Peefy) [inline kclvm/runner/src/linker.rs:21]: Do not delete these in this PR. The wasm target will need to use these later.

File: kclvm/runner/src/command.rs [hunk 3]
```diff
@@ -152,6 +63,51 @@ impl Command {
...
+        libs: &[String],
+        lib_path: String,
+        cmd: &mut std::process::Command,
+    ) {
+        cmd.args(libs)
+            .arg("kclvm.lib")
+            .arg("/link")
+            .arg("/ENTRY:kclvm_main")
+            .arg("/NOLOGO")
+            .arg(format!(r#"/LIBPATH:"{}\_output""#, self.executable_root))
```
Comment (Peefy) [inline kclvm/runner/src/command.rs:101]: What does the path `{executable_root}\_output` mean? Do the windows building artifact path? Will this path appear in the package for users?

File: scripts/build-windows/build_kclvm_cli.bat [hunk 4]
```diff
@@ -0,0 +1,30 @@
+:: Copyright 2021 The KCL Authors. All rights reserved.
+
+setlocal
+cd %~dp0
+
+:: install kclvm-runtime
```
Comment (Peefy) [inline scripts/build-windows/build_kclvm_cli.bat:6]: The kclvm-runtime needs not to build separately. Ref: https://github.com/KusionStack/KCLVM/pull/343

File: scripts/build-windows/build.bat [hunk 5]
```diff
@@ -10,72 +10,38 @@ go run unzip.go
 go run gen_pth.go
 
 :: renname
-go run rename.go -old="_output\kclvm-windows\python.exe" -new="_output\kclvm-windows\kclvm.exe"
+go run rename.go -old="_output\python.exe" -new="_output\kclvm_py.exe"
```
Comment (Peefy) [inline scripts/build-windows/build.bat:13]: Why using `kclvm_py` instead of kclvm

File: scripts/build-windows/build.bat [hunk 6]
```diff
@@ -10,72 +10,38 @@ go run unzip.go
...
+go build -o .\_output\kcl-fmt.exe    kcl-fmt.go
+go build -o .\_output\kcl-plugin.exe kcl-plugin.go
+go build -o .\_output\kcl-vet.exe    kcl-vet.go
 
 :: run hello.k
-_output\kclvm-windows\kclvm.exe -m kclvm ..\..\hello.k
-_output\kclvm-windows\kcl-go.exe run     ..\..\hello.k
-_output\kclvm-windows\kcl.exe            ..\..\hello.k
+_output\kcl.exe            ..\..\hello.k
+_output\kclvm-cli.exe run ..\..\samples\hello_invalid.k
```
Comment (Peefy) [inline scripts/build-windows/build.bat:45]: Why run an invalid case here?

File: scripts/build-windows/build.bat [hunk 7]
```diff
@@ -10,72 +10,38 @@ go run unzip.go
...
+go build -o .\_output\kcl-lint.exe   kcl-lint.go
+go build -o .\_output\kcl-fmt.exe    kcl-fmt.go
+go build -o .\_output\kcl-plugin.exe kcl-plugin.go
+go build -o .\_output\kcl-vet.exe    kcl-vet.go
 
 :: run hello.k
-_output\kclvm-windows\kclvm.exe -m kclvm ..\..\hello.k
-_output\kclvm-windows\kcl-go.exe run     ..\..\hello.k
-_output\kclvm-windows\kcl.exe            ..\..\hello.k
+_output\kcl.exe            ..\..\hello.k
```
Comment (Peefy) [inline scripts/build-windows/build.bat:44]: Format the bat script.

File: kclvm/runner/src/lib.rs [hunk 8]
```diff
@@ -201,8 +201,12 @@ pub fn execute(
     );
     let result = runner.run(args);
 
-    // Clean temp files
+    // Clean temp files.
+    // FIXME: On windows, sometimes there will be an error that the file cannot be accessed.
```
Comment (Peefy) [inline kclvm/runner/src/lib.rs:205]: Tracking an issue number directly.

File: samples/hello_invalid.k (hunk 1 above)
Comment (zong-zhe) [reply samples/hello_invalid.k]: fixed.

File: scripts/build-windows/build.bat (hunk 6 above)
Comment (zong-zhe) [reply scripts/build-windows/build.bat]: The invalid case removed.

File: scripts/build-windows/build_kclvm_cli.bat (hunk 4 above)
Comment (zong-zhe) [reply scripts/build-windows/build_kclvm_cli.bat]: fixed.

File: scripts/build-windows/build.bat (hunk 7 above)
Comment (zong-zhe) [reply scripts/build-windows/build.bat]: fixed.

File: kclvm/runner/src/lib.rs (hunk 8 above)
Comment (zong-zhe) [reply kclvm/runner/src/lib.rs]: fixed.

File: kclvm/runner/src/command.rs (hunk 3 above)
Comment (Peefy) [reply kclvm/runner/src/command.rs:161]: > 

Put all kcl windows artifacts into the `bin` directory.

Comment (zong-zhe) [reply kclvm/runner/src/command.rs]: fixed

**Review by Peefy — DISMISSED** [review]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**